client/src/components/PortfolioFilters.tsx

```diff
@@ -0,0 +1,394 @@
+    const { ownernames: ownernamesSelections, zip: zipSelections } = filterContext.filterSelections;
+
+    const [rsunitslatestActive, setRsunitslatestActive] = React.useState(false);
+    const updateRsunitslatest = () => {
+      setRsunitslatestActive(!rsunitslatestActive);
+      setFilterContext({
+        ...filterContext,
+        filterSelections: {
+          ...filterContext.filterSelections,
+          rsunitslatest: !filterContext.filterSelections.rsunitslatest,
+        },
+      });
+    };
+
+    const [ownernamesActive, setOwnernamesActive] = React.useState(false);
+    const [ownernamesIsOpen, setOwnernamesIsOpen] = React.useState(false);
+    const onOwnernamesApply = (selectedList: any) => {
+      setOwnernamesActive(!!selectedList.length);
+      setOwnernamesIsOpen(false);
+      setFilterContext({
+        ...filterContext,
+        filterSelections: {
+          ...filterContext.filterSelections,
+          ownernames: selectedList,
+        },
+      });
+    };
+
+    const [unitsresActive, setUnitsresActive] = React.useState(false);
+    const [unitsresIsOpen, setUnitsresIsOpen] = React.useState(false);
+    const onUnitsresApply = (selectedList: any) => {
+      console.log({ onUnitsresApply: selectedList });
+      setUnitsresActive(selectedList !== MINMAX_DEFAULT);
+      setUnitsresIsOpen(false);
+      setFilterContext({
+        ...filterContext,
+        filterSelections: {
+          ...filterContext.filterSelections,
+          unitsres: selectedList,
+        },
+      });
+    };
+
+    const [zipActive, setZipActive] = React.useState(false);
+    const [zipIsOpen, setZipIsOpen] = React.useState(false);
+    const onZipApply = (selectedList: any) => {
+      setZipActive(!!selectedList.length);
+      setZipIsOpen(false);
+      setFilterContext({
+        ...filterContext,
+        filterSelections: {
+          ...filterContext.filterSelections,
+          zip: selectedList,
+        },
+      });
+    };
+
+    const clearFilters = () => {
+      setRsunitslatestActive(false);
+      setOwnernamesActive(false);
+      setUnitsresActive(false);
+      setZipActive(false);
+      setFilterContext({
+        ...filterContext,
+        filterSelections: {
+          rsunitslatest: false,
+          ownernames: [],
+          unitsres: MINMAX_DEFAULT,
+          zip: [],
+        },
+      });
+    };
+
+    return (
+      <div className="PortfolioFilters" ref={ref}>
+        <div className="filter-for">
+          <span className="pill-new">
+            <Trans>New</Trans>
+          </span>
+          <Trans>Filters</Trans>:
+        </div>
+        <div className="filters-container">
+          <div className="filters">
+            <button
+              aria-pressed={rsunitslatestActive}
+              onClick={updateRsunitslatest}
+              className="filter-toggle"
+            >
+              <div className="checkbox">{rsunitslatestActive && <CheckIcon />}</div>
```

Comment:
noting for future: we might want to look into making this a checkbox element for accessibility

Reply:
I was wondering that at first, but checking with tahnee my understanding is that it's purely aesthetic - the whole surrounding box is a button and clicking anywhere on it is supposed to trigger the toggle, so I was treating it like a toggle button (like from that [a11y toggle article](https://joshcollinsworth.com/blog/accessible-toggle-buttons) tahnee shared a while back)